dthm4kaiako/templates/events/event_detail.html

```diff
@@ -105,11 +125,55 @@ <h3 id="event-schedule" class="event-heading">Schedule</h3>
                 {% endif %}
             </nav>
 
-            {% if event.registration_link and not event.has_ended %}
+        {% if event.is_cancelled %} 
+            <div style="padding-top: 20px;">
+                <label style="justify-content: center; display: flex; background-color:rgb(255, 198, 198); color:black; font-size: medium; border-radius: 8px; padding-left: 12px; padding-right: 12px; padding-top: 12px; padding-bottom: 12px;">
+                    <h5><b>Cancelled</b></h5>
+                </label>
+            </div>
+        {% else %}
+            {% if event.external_event_registration_link and not event.has_ended %}
                 <hr>
-                <a class="btn btn-success btn-block btn-lg" href="{{ event.registration_link }} " role="button">
+                <a class="btn btn-success btn-block btn-lg" href="{{ event.external_event_registration_link }} " role="button">
                     {{ event.get_registration_type_display }}
-                </a>
+                </a>                
+            {% elif not event.has_ended and event.is_register_or_apply %}
+
+                {% if has_user_applied %}
+                    <hr>
+                    <a class="btn btn-success btn-block btn-lg" href="{% url 'events:register' event.pk %}" role="button">
+                        Update my details
+                    </a>
+                    <hr>
+                    
+                    <!-- Button trigger modal -->
+                    <button type="button" class="btn btn-warning btn-block btn-lg" data-toggle="modal" data-target="#event-details-withdraw-modal-{{registration_pk}}">
+                        Withdraw
+                    </button>
+
+                {% else %}
+
+                    {% if event.is_less_than_one_week_prior_event and event.is_catered %}
+                    <div style="padding-top: 20px;">
+                        <label style="justify-content: center; display: flex; background-color:rgb(209, 209, 209); color:black; font-size: medium; border-radius: 8px; padding-left: 12px; padding-right: 12px; padding-top: 12px; padding-bottom: 12px;">
```

Comment:
Is there any reason why inline styles are used here compared to writing a class for this?

Reply:
I'll get onto this one tomorrow.

Reply:
All of the labels' and badges' styling have been moved to classes in the style sheet.